dvc/lock.py

```diff
@@ -148,9 +148,8 @@ def __init__(
     ):  # pylint: disable=super-init-not-called
         import socket
 
-        super().__init__(lockfile)
-
         self._tmp_dir = tmp_dir
+        super().__init__(lockfile)
```

Comment:
`flufl.lock.__init__()` will end up calling out our `_set_claimfile()` where we reference `_tmp_dir`, so it needs to be set before calling the super `__init__` chain